try/run_central_instances.py

```diff
@@ -0,0 +1,2341 @@
+            if not isinstance(nick, str):
+                raise ServerDefinitionFileError(
+                    "Item {0!r} in server group {1!r} in WBEM server "
+                    "definition file {2!r} must be a string, but is a {3}".
+                    format(nick, sg_nick, self._filename, type(nick)))
+            if nick in server_groups:
+                self._check_sg(nick, server_groups, servers, visited_sg_nicks)
+            elif nick not in servers:
+                raise ServerDefinitionFileError(
+                    "Item {0!r} in server group {1!r} in WBEM server "
+                    "definition file {2!r} is not a known server or server "
+                    "group".
+                    format(nick, sg_nick, self._filename))
+
+    @property
+    def filename(self):
+        """
+        Path name of the WBEM server definition file.
+        """
+        return self._filename
+
+    def get_server(self, nickname):
+        """
+        Return a `ServerDefinition` object for the server with the specified
+        nickname.
+        """
+        try:
+            server_dict = self._servers[nickname]
+        except KeyError:
+            raise ValueError(
+                "Server nickname {0!r} not found in WBEM server definition "
+                "file {1!r}".
+                format(nickname, self._filename))
+        return ServerDefinition(nickname, server_dict)
+
+    def list_all_servers(self):
+        """
+        return list of all actual servers.
+        """
+        return [self.get_server(nickname) for nickname in self._servers]
+
+    def list_servers(self, nicknames=None):
+        """
+        Iterate through the servers of the server group with the specified
+        nicknames, or the single server with the specified nickname, and yield
+        a `ServerDefinition` object for each server.
+
+        nicknames None, string defining a nickname or list of nicknames
+        """
+        if not nicknames:
+            return self.list_all_servers()
+
+        if isinstance(nicknames, six.string_types):
+            nicknames = [nicknames]
+
+        sd_list = []
+        sd_nick_list = []
+        for nickname in nicknames:
+            if nickname in self._servers:
+                sd_list.append(self.get_server(nickname))
+            elif nickname in self._server_groups:
+                for item_nick in self._server_groups[nickname]:
+                    for sd in self.list_servers(item_nick):
+                        if sd.nickname not in sd_nick_list:
+                            sd_nick_list.append(sd.nickname)
+                            sd_list.append(sd)
+            else:
+                raise ValueError(
+                    "Server group or server nickname {0!r} not found in WBEM "
+                    "server definition file {1!r}".
+                    format(nickname, self._filename))
+        return sd_list
+
+
+class ServerDefinition(object):
```

Comment:
I don't think the `ServerDefinition` class should be copied, but instead imported.

Reply:
Done.  Note that I added the default_namespace to this class.  We had discussed this and agreed that there was really no need for it in the tests.  However, if this class is to be more general, there may be a need for this as an optional property simply because it it a part of WBEMConnection and useful in writing other scripts using the ServerDefinition object.